worldedit-core/src/main/java/com/sk89q/worldedit/command/SchematicCommands.java

```diff
@@ -611,6 +719,56 @@ public Void call() throws Exception {
             Transform transform = holder.getTransform();
             Clipboard target;
 
+            boolean checkFilesize = false;
+
+            if (Settings.IMP.PATHS.PER_PLAYER_SCHEMATICS
+                && Settings.IMP.EXPERIMENTAL.PER_PLAYER_FILE_SIZE_LIMIT > -1) {
+                checkFilesize = true;
+            }
+
+            double directorysizeKb = 0;
+            String curFilepath = file.getAbsolutePath();
+            final String SCHEMATIC_NAME = file.getName();
+
+            double oldKbOverwritten = 0;
+            String overwrittenPath = curFilepath;
+
+            if (checkFilesize) {
+                File parentDir = new File(file.getParent());
+
+                for (File child : parentDir.listFiles()) {
+                    if (child.getName().endsWith(".schem") || child.getName().endsWith(".schematic")) {
+                        directorysizeKb += Files.size(Paths.get(child.getAbsolutePath())) / 1000.0;
+                    }
+                }
+
+
+                if (overwrite) {
+                    oldKbOverwritten = Files.size(Paths.get(file.getAbsolutePath())) / 1000.0;
+                    int iter = 1;
+                    while (new File(overwrittenPath + "." + iter + "." + format.getPrimaryFileExtension()).exists()) {
+                        iter++;
+                    }
+                    file = new File(overwrittenPath + "." + iter + "." + format.getPrimaryFileExtension());
+                }
+            }
+
+
+            if (Settings.IMP.PATHS.PER_PLAYER_SCHEMATICS && Settings.IMP.EXPERIMENTAL.PER_PLAYER_FILE_NUM_LIMIT > -1) {
+
+                int curFilesAmount = new File(file.getParent()).listFiles().length;
+                int limit = Settings.IMP.EXPERIMENTAL.PER_PLAYER_FILE_NUM_LIMIT;
+
+                if (curFilesAmount >= limit) {
+                    TextComponent noSlotsErr = TextComponent.of( //TODO - to be moved into captions/translatablecomponents
+                        "You have " + String.format("You have " + curFilesAmount + "/" + limit + " saved schematics. Delete some to save this one!",
```

Comment:
"You have" + "You have".
Plus this sounds a bit misleading. I'd leave a note about the schematic file size limit.

Reply:
> "You have" + "You have".
good catch ;-;

> Plus this sounds a bit misleading. I'd leave a note about the schematic file size limit.
misleading how? I'm not sure I understand